codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2LocalFlowController.java

```diff
@@ -65,14 +65,26 @@ public DefaultHttp2LocalFlowController(Http2Connection connection,
         connection.addListener(new Http2ConnectionAdapter() {
             @Override
             public void onStreamAdded(Http2Stream stream) {
-                stream.localFlowState(new DefaultFlowState(stream, 0));
+                // Unconditionally used the reduced flow control state because it requires no object allocation
+                // and the DefaultFlowState will be allocated in onStreamActive.
+                stream.localFlowState(REDUCED_FLOW_STATE);
             }
 
             @Override
             public void onStreamActive(Http2Stream stream) {
                 // Need to be sure the stream's initial window is adjusted for SETTINGS
                 // frames which may have been exchanged while it was in IDLE
-                state(stream).window(initialWindowSize);
+                stream.localFlowState(new DefaultFlowState(stream, initialWindowSize));
+            }
+
+            @Override
+            public void onStreamEligibleForRemove(Http2Stream stream) {
+                // If the stream is now eligible for removal, but will persist in the priority tree then we can
```

Comment:
I don't think we need this event.  Capturing the closed event should be sufficient.